cpp/src/arrow/acero/exec_plan.cc

```diff
@@ -358,10 +359,38 @@ std::optional<int> GetNodeIndex(const std::vector<ExecNode*>& nodes,
   return std::nullopt;
 }
 
+const std::string kAceroAlignmentHandlingEnvVar = "ACERO_ALIGNMENT_HANDLING";
```

Comment:
C-style string as suggested by the linter is better for constants. And that's what `GetEnvVar` takes under-the hood, so a heap-alloc for std::string never happens.

Reply:
Fair enough.  I only saw the string accepting variant and missed the char* version.  I've switched.